crates/kornia-core/src/storage.rs

```diff
@@ -51,16 +76,21 @@ where
     /// A new tensor storage if successful, otherwise an error.
     pub fn new(len: usize, alloc: A) -> Result<Self, TensorAllocatorError> {
         // allocate memory for tensor storage
-        let ptr =
-            alloc.alloc(Layout::array::<T>(len).map_err(TensorAllocatorError::LayoutError)?)?;
+        let layout = Layout::array::<T>(len).map_err(TensorAllocatorError::LayoutError)?;
+        let ptr = alloc.alloc(layout)?;
+        let owner = TensorCustomAllocationOwner {
+            alloc: alloc.clone(),
```

Comment:
I cloned because I needed alloc below also, to store in TensorStorage struct. (I know it's not ideal to have two copies of it but I couldn't easily find a way around it.). Do you mean wrap A in Arc everywhere or just here?

Reply:
yeap, wrap `alloc: Arc<A>` in both `TensorStorage` and `TensorCustomAllocationOwner` maybe ?

Reply:
Done